flake.nix

```diff
@@ -72,7 +72,10 @@
                     # LIBCLANG_PATH points rocksdb to a clang.so on Linux
                     export LIBCLANG_PATH="$(readlink -f ${pkgs.clang}/resource-root/include | xargs dirname | xargs dirname | xargs dirname)"
 
-                    cowsay "Snowbridge Dev Environment"
+                    echo "Initializing Dev Environment..."
+                    ./init.sh
```

Comment:
Could you elaborate some more abou benefit of inline script here?  IMHO seperate to another shell script is more maintainable and yes the script here applies also for none nix env

Reply:
`init.sh` looks like it's only `forge install` and `pnpm install`, which are dev env setup commands that aren't directly related to Nix. We could run them outside Nix, but we're not planning to afaik.

I can't see us gaining much from a separate script here. We could separate Nix-related commands in `shellHook` vs non-Nix-related commands in `init.sh`, but that only saves us a small amount of time if we decide to stop using Nix.

Reply:
Agreed about `husky` 👌 It's possible, though unlikely, that we'll want to run the non-`husky` commands in start-services. This would allow start-services to work in the same Nix shell after a "clean" operation (eg. deleting `node_modules/` and the submodules in `lib/`).

Assuming we use `git clean` for the clean action, this won't delete the Foundry submodules. So adding `forge install` to start-services is less of a concern.

This leaves the `pnpm install` and `hack-ethereum` commands, which are necessary after removing node_modules and before starting the test environment. A hack for this is restarting the Nix shell: with `direnv` set up, `cd ~ && cd -` will rerun the shellHook, including `pnpm install` and the postinstall script.

If we go with this approach, let's add a note to the README about restarting the Nix shell after a `git clean`.